lib/puppet/face/config.rb

```diff
@@ -94,16 +96,47 @@
 
         values = Puppet.settings.values(Puppet[:environment].to_sym, options[:section].to_sym)
         report_section_and_environment(options[:section], Puppet.settings[:environment])
-        if args.length == 1
-          puts values.interpolate(args[0].to_sym)
-        else
-          args.sort.each do |setting_name|
-            puts "#{setting_name} = #{values.interpolate(setting_name.to_sym)}"
-          end
+
+        to_be_rendered = {}
+        args.sort.each do |setting_name|
+          to_be_rendered[setting_name] = values.interpolate(setting_name.to_sym)
         end
       end
-      nil
+
+      # convert symbols to strings before formatting output
+      if render_all_settings
+        to_be_rendered = stringifyhash(to_be_rendered)
+      end
+      to_be_rendered
+    end
+
+    when_rendering :console do |to_be_rendered|
+      output = ''
+      if to_be_rendered.keys.length > 1
+        to_be_rendered.keys.sort.each do |setting|
+          output << "#{setting} = #{to_be_rendered[setting]}\n"
+        end
+      else
+        output << "#{to_be_rendered.to_a[0].last}\n"
+      end
+
+      output
+    end
+  end
+
+  def stringifyhash(hash)
+    newhash = {}
+    hash.each do |key, val|
+      key = key.to_s
+      if val.is_a? Hash
+        newhash[key] = stringifyhash(val)
+      elsif val.is_a? Symbol
+        newhash[key] = val.to_s
+      else
```

Comment:
Can `val` ever be an `Array` (thinking of settings like `:supported_checksum_types` that are of type `:array`)? If so, do we need to join with a comma here?

Reply:
val can be an array but arrays can only contain strings in settings, and we don't need to join them with a comma because network format handlers all take care of this.